test/e2e/framework/kubernetes/get-external-crd.go

```diff
@@ -0,0 +1,62 @@
+package kubernetes
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+)
+
+func downloadExternalCRDs(chartPath string) error {
+	crdUrls := []string{
+		"https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/main/example/prometheus-operator-crd/monitoring.coreos.com_servicemonitors.yaml",
+	}
+
+	for _, crdUrl := range crdUrls {
+		crd, err := fetchYAML(crdUrl)
+		if err == nil && crd != nil {
+			crdName, err := extractFileName(crdUrl)
+			if err == nil && crdName != "" {
+				log.Printf("CRD exists %s", crdName)
+				log.Printf("File path to be written to %s", filepath.Join(chartPath, "/crds/"+crdName))
+				err = saveToFile(filepath.Join(chartPath, "/crds/"+crdName), crd)
+				if err != nil {
+					return fmt.Errorf("failed to save crd.yaml /crds dir: %w", err)
+				}
+			}
+		}
+		if err != nil {
```

Comment:
this block should be straight at the start, just after `crd, err := fetchYAML(crdUrl)` - (relates to previous comments)